client/state/sites/selectors.js

```diff
@@ -313,12 +313,7 @@ export function isSitePreviewable( state, siteId ) {
  * @return {*}  The value of that option or null
  */
 export function getSiteOption( state, siteId, optionName ) {
-	const site = getRawSite( state, siteId );
-	if ( ! site || ! site.options ) {
-		return null;
-	}
-
-	return site.options[ optionName ];
+	return get( getSiteOptions( state, siteId ), optionName, null );
```

Comment:
While I like the change to return `null` instead of `undefined` here, I'm hesitant to land it, because this might break something that relies on this difference. Have we gone through all of the `getSiteOption` usages for this?

Reply:
Hi @tyxla, Before we found a case where this difference was problematic and a PR that solves it was already merged. I checked again and it looks like the usages are safe (most have a boolean check to see if the option exist). There are many direct usages and even more indirect dependencies of the result (which I tried to follow), but there is always some risk that I missed a strange case :(
I also executed e2e tests and for the desktop resolution, only the insert payment button test case failed, but I was able to manually insert the button with success. This test case also fails on my environment in other branches.

Reply:
I've done some manual tests of potentially problematic locations that could handle `undefined` and `null` differently, didn't find any issues. Looks good to me, let's go with it.